File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.cpp
#include <vector>
#include <set>
using namespace std;

set<int> s;

int solution(vector<int> nums)
{
    for(int num:nums){
        s.insert(num);
    }

    if(nums.size()/2 < s.size()) return nums.size()/2;
    else return s.size();
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.HashSet;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        HashSet<Integer> numSet = new HashSet<>();
        
        int max_val = nums.length / 2;
        
        for(int num : nums){
            numSet.add(num);
        }
        
        if(numSet.size() > max_val){
            return max_val;
        }
        
        return numSet.size();
    }
}